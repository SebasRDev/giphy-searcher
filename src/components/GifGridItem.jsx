import React from 'react'
import PropTypes from 'prop-types'

const GifGridItem = ({title,url}) => {
  console.log('cards')
  return (
    <div className='card animate__animated animate__backInUp'>
      <img src={url} alt={title}/>
      <p>{title}</p>
    </div>
  )
}

export default GifGridItem

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}