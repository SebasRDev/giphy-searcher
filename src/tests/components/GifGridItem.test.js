import {shallow} from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem/>', () => {
  
  const title = "Un titulo"
  const url = "https://localhost/algo.jpg"
  const wrapper = shallow(<GifGridItem title={title} url={url}/>)

  test('Should show component correctly', ()=>{
    expect(wrapper).toMatchSnapshot();
  })

  test('should have a paragraph with the title', ()=>{
    const p = wrapper.find('p');
    expect( p.text().trim() ).toBe( title );
  })

  test('should have th img equal to url and alt od props', () => {
    const img = wrapper.find('img');
    expect( img.props().src ).toBe( url );
    expect( img.props().alt ).toBe( title );
  })

  test('should have animate__backInUp', ()=>{
    const div = wrapper.find('div');
    const className = div.props().className;

    expect( className.includes('animate__backInUp') ).toBe(true);
  })

})