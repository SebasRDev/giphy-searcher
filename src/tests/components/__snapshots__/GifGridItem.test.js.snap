// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Should show component correctly 1`] = `
<div
  className="card animate__animated animate__backInUp"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
